src/switchcraft_advanced/services/intune_service.py

```
@@ -0,0 +1,362 @@
+
+    def authenticate(self, tenant_id, client_id, client_secret):
+        """Authenticates with MS Graph using Client Credentials."""
+        url = f"https://login.microsoftonline.com/{tenant_id}/oauth2/v2.0/token"
+        data = {
+            "client_id": client_id,
+            "scope": "https://graph.microsoft.com/.default",
+            "client_secret": client_secret,
+            "grant_type": "client_credentials"
+        }
+        try:
+            resp = requests.post(url, data=data)
```

Ruff (0.14.8)

Probable use of `requests` call without timeout

(S113)